src/cmd/linuxkit/azure.go

```diff
@@ -0,0 +1,372 @@
+	publicIPAddressesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	interfacesClient = network.NewInterfacesClient(subscriptionID)
+	interfacesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	virtualMachinesClient = compute.NewVirtualMachinesClient(subscriptionID)
+	virtualMachinesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+}
+
+func getOrCreateResourceGroup(resourceGroupName, location string) *resources.Group {
+	var resourceGroup resources.Group
+	resourceGroup, err := groupsClient.Get(resourceGroupName)
+	if err != nil {
+		log.Fatalf("\nError in getting resource group")
+	}
+	if &resourceGroup != nil {
+		return &resourceGroup
+	}
+
+	return createResourceGroup(resourceGroupName, location)
+}
+
+func createResourceGroup(resourceGroupName, location string) *resources.Group {
+	fmt.Printf("\nCreating resource group in %s", location)
+
+	resourceGroupParameters := resources.Group{
+		Location: &location,
+	}
+	group, err := groupsClient.CreateOrUpdate(resourceGroupName, resourceGroupParameters)
+	if err != nil {
+		fmt.Println(err.Error())
+		log.Fatalf("Unable to create resource group")
+	}
+
+	return &group
+}
+
+func createStorageAccount(accountName, location string, resourceGroup resources.Group) {
+	fmt.Printf("\nCreating storage account in %s, resource group %s\n", location, *resourceGroup.Name)
+
+	storageAccountCreateParameters := storage.AccountCreateParameters{
+		Sku: &storage.Sku{
+			Name: storage.StandardLRS,
+		},
+		Location: &location,
+		AccountPropertiesCreateParameters: &storage.AccountPropertiesCreateParameters{},
+	}
+
+	storageChannel, errorChannel := accountsClient.Create(*resourceGroup.Name, accountName, storageAccountCreateParameters, nil)
+	for {
+		select {
+		case _, ok := <-storageChannel:
+			if !ok {
+				storageChannel = nil
+			}
+		case _, ok := <-errorChannel:
+			if !ok {
+				errorChannel = nil
+			}
+		}
+		if storageChannel == nil && errorChannel == nil {
+			break
+		}
+	}
+}
+
+func uploadVMImage(resourceGroupName string, accountName string, imagePath string) {
+	accountKeys, err := accountsClient.ListKeys(resourceGroupName, accountName)
+	if err != nil {
+		fmt.Println(err.Error())
+		log.Fatalf("Unable to retrieve storage account key")
+	}
+
+	keys := *(accountKeys.Keys)
+
+	absolutePath, err := filepath.Abs(imagePath)
+	if err != nil {
+		fmt.Println(err.Error())
+		log.Fatalf("Unable to get absolute path")
+	}
+
+	directory, image := filepath.Split(absolutePath)
+
+	dockerMount := fmt.Sprintf("%s:/vhds", directory)
+	storageAccountNameArg := fmt.Sprintf("STORAGE_ACCOUNT_NAME=%s", accountName)
+	storageAccountKeyArg := fmt.Sprintf("STORAGE_ACCOUNT_KEY=%s", *keys[0].Value)
+	vhdPath := fmt.Sprintf("VHD_PATH=/vhds/%s", image)
+
+	output, err := exec.Command("docker", "run", "-v", dockerMount, "-e", vhdPath, "-e", storageAccountNameArg, "-e", storageAccountKeyArg, "radumatei/azure-vhd-upload:alpine").CombinedOutput()
```

Comment:
Here is the thing I would like you to review: there is already a very competent tool that uploads VHDs to Azure. Because the Go SDK changed in the meantime, it meant I should have had to add two versions of it. Didn't want to do it, so I think this is the easiest way for the moment. 

For the first version, until the SDK stabilizes, do you think we can have this way of uploading the image?